examples/C++/HelloWorldExampleTCP/HelloWorld_main.cpp

```diff
@@ -43,24 +47,46 @@ int main(int argc, char** argv)
             if (argc >= 3)
             {
                 count = atoi(argv[2]);
-                if (argc == 4)
+                if (argc > 3)
                 {
                     sleep = atoi(argv[3]);
                 }
+
+				if (argc > 4)
+				{
+					wan_ip = std::string(argv[4]);
+				}
+
+				if (argc > 5)
+				{
+					port = atoi(argv[5]);
+				}
             }
         }
         else if(strcmp(argv[1],"subscriber")==0)
         {
             type = 2;
+			
+			if (argc > 2)
+			{
+				wan_ip = std::string(argv[2]);
+			}
+
+			if (argc > 3)
+			{
+				port = atoi(argv[3]);
+			}
         }
     }
     else
     {
         std::cout << "publisher OR subscriber argument needed" << std::endl;
-        std::cout << "publisher has optional arguments: publisher [times] [interval] " << std::endl;
+        std::cout << "publisher has optional arguments: publisher [times] [interval] [wan_ip] [port] " << std::endl;
         std::cout << "\ttimes: Number of messages to send (default: unlimited = 0). " << std::endl;
         std::cout << "\t\tIf times is set greater than 0, no messages will be sent until a subscriber matches. " << std::endl;
-        std::cout << "\tinterval: Milliseconds between messages (default: 100). " << std::endl;
+		std::cout << "\tinterval: Milliseconds between messages (default: 100). " << std::endl;
+		std::cout << "\twap_ip: TODO " << std::endl;
+		std::cout << "\tport: TODO " << std::endl;
```

Comment:
Please, change tabs to 4 spaces, and add the description of the fields. Remember in this case, the publisher will act as TCP server, so the IP is used iss WAN address and the subscriber will act as TCP client, so the IP must be the remote public address.

Reply:
Updated.